File: VB/ColorMappingExample/Model/Category.vb
Imports DevExpress.Mvvm.POCO

Namespace ColorMappingExample.Model
    Public Class Category
        Private _colorData As Object

        Protected Sub New(ByVal id As Integer, ByVal caption As String, ByVal color As Object)
            Me.Id = id
            Me.Caption = caption
            ColorData = color
        End Sub
        Public Shared Function Create(ByVal id As Integer, ByVal caption As String, ByVal color As Object) As Category
            Return ViewModelSource.Create(Function() New Category(id, caption, color))
        End Function

        Protected Sub New()
        End Sub

        Public Overridable Property Id() As Integer
        Public Overridable Property Caption() As String
        Public Overridable Property ColorData() As Object
            Get
                Return _colorData
            End Get
            Set(ByVal value As Object)
                _colorData = value
                ColorDataInfo = "Saved as " & value.GetType().ToString() & " value= " & value.ToString()
            End Set
        End Property
        Public Overridable Property ColorDataInfo() As String
    End Class
End Namespace

File: CS/ColorMappingExample/Model/Category.cs
using DevExpress.Mvvm.POCO;

namespace ColorMappingExample.Model {
    public class Category {
        object _colorData;

        protected Category(int id, string caption, object color) {
            Id = id;
            Caption = caption;
            ColorData = color;
        }
        public static Category Create(int id, string caption, object color) {
            return ViewModelSource.Create(() => new Category(id, caption, color));
        }

        protected Category() { }

        public virtual int Id { get; set; }
        public virtual string Caption { get; set; }
        public virtual object ColorData {
            get {
                return _colorData;
            }
            set {
                _colorData = value;
                ColorDataInfo = "Saved as " + value.GetType().ToString() + " value= " + value.ToString();
            }
        }
        public virtual string ColorDataInfo { get; set; }
    }
}
